### ISSUE TITLE
COSE_Sign1 header with simple type key should be rejected

### ISSUE THREAD

**Issue by qmuntal** [veraison/go-cose#38]
Found via fuzzing.

### What did you do?

```go
package main

import "github.com/veraison/go-cose"

func main() {
	b := []byte("҄҄C\xa100\xa1\xf30T00000000000000000000A0000000")
	var msg cose.Sign1Message
	err := msg.UnmarshalCBOR(b)
	if err == nil {
		panic("error expected!")
	}
}
```

### What did you expect to see?

UnmarshalCBOR returns an error due to malformed COSE_Sign1 message

### What did you see instead?

UnmarshalCBOR does not return an error.

Notice that the `[]byte("҄C\xa100\xa100T00000000000000000000\xcbA0000000000")` input has an invalid unprotected map key. It is a simple value, when it should be an integer or a string. as per [RFC8152, Section 3](https://datatracker.ietf.org/doc/html/rfc8152#section-3)

**Comment by shizhMSFT** [comment]
The unprotected map key `\xf3` is a simple value `19` of major type `7`, which is an unsigned value (see [RFC 8949  3.3](https://datatracker.ietf.org/doc/html/rfc8949#section-3.3)). Since the underlying decoder `github.com/fxamacker/cbor/v2` considers simple values `0..19, 24` as `uint` (see [code](https://github.com/fxamacker/cbor/blob/v2.4.0/decode.go#L1005)), we are not able to verify the map keys whether they are int / tstr or others at the application level in an easy way.

**Comment by shizhMSFT** [comment]
Addressed by #39
